static/app/views/alerts/issueRuleEditor/sentryAppRuleModal.tsx

```diff
@@ -0,0 +1,39 @@
+import {Component, Fragment} from 'react';
+
+import {ModalRenderProps} from 'app/actionCreators/modal';
+import {tct} from 'app/locale';
+import SentryAppExternalForm, {
+  Config,
+} from 'app/views/organizationIntegrations/sentryAppExternalForm';
+
+type Props = ModalRenderProps & {
+  sentryAppInstallationUuid: string;
+  appName: string;
+  config: Config;
+  action: 'create' | 'update';
+  onSubmitSuccess: (...params: any[]) => void;
+};
+
+class SentryAppRuleModal extends Component<Props> {
```

Comment:
This can be a functional component :)

```tsx
const SentryAppRuleModal = ({
  Header,
  Body,
  sentryAppInstallationUuid,
  appName,
  config,
  action,
}: Props) => (
  <Fragment>
    <Header closeButton>{tct('[name] Settings', {name: appName})}</Header>
    <Body>
      <SentryAppExternalForm
        sentryAppInstallationUuid={sentryAppInstallationUuid}
        appName={appName}
        element="alert-rule-action"
        config={config}
        action={action}
        onSubmitSuccess={this.props.onSubmitSuccess}
        // TODO(leander): Add new defaulting fields for alerts
      />
    </Body>
  </Fragment>
);
```

Reply:
changed, thanks :)